File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/filterCodes/model/GetFilterCodes.java
package org.ngbp.jsonrpc4jtestharness.rpc.filterCodes.model;

import java.util.List;

public class GetFilterCodes {
    public List<Filters> filters;
    public static class Filters{
        public Integer filterCode;
        public String expires;

        @Override
        public String toString() {
            return "{ \"filterCode\": "+filterCode+", \"expires\": \""+expires+"\" }";
        }
    }

    @Override
    public String toString() {
        return "{\"jsonrpc\": \"2.0\","+
                "\"result\": {"+
                "\"filters=\"" + getListToString() +
                '}';
    }

    private String getListToString() {
        String array ="[";
            if(filters==null && filters.size()==0){
                return array+"]";
            }else{
                for(int i=0;i<filters.size();i++){
                    array = array+filters.get(i);
                    if(i!=filters.size()-1){
                        array = array+",";
                    }
                }
                return array+"]";
            }
    }
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/filterCodes/model/GetFilterCodes.kt
package org.ngbp.jsonrpc4jtestharness.rpc.filterCodes.model

import org.ngbp.jsonrpc4jtestharness.rpc.RpcResponse

data class GetFilterCodes(
        var filters: MutableList<Filters?>? = null
) : RpcResponse()

data class Filters(
        var filterCode: Int? = null,
        var expires: String? = null
)
